File: Problem Solving/C/perfect.c
/*check whether the given number is perfect number or not*/
#include<stdio.h>
#include<conio.h>
void main()
{
int sum=0,num;
print("Enter your number");
scanf("%d",&num);
for(int i=1;i<num;i++)
{
if(num%i==0)
{
sum=sum+i;
}
}
if(sum==num)
{
printf("%d is the perfect number",num);
}
else
{
printf("%d is not the perfect number",num);
}
getch();
}

File: Problem Solving/Java Programming/perfect.java
//Java Program to check Whether a given Number is Perfect number or not

import java.util.*;
class perfect{
public static void main(String args[])
{
Scanner sc = new Scanner(System.in);
System.out.println("Enter your Number");
int number = sc.nextInt();
int sum = 0;
for(int i=1;i<number;i++)
{
if(number%i == 0)
{
sum = sum+i;
}
}
if(sum==number)
{
System.out.println(number+" is the Perfect Number");
}
else
{
System.out.println(number+" is not the Perfect Number");
}
}
}